### PR TITLE
samples: power: add check for PM debug hooks, refactor power_mgr sample

### PR DESCRIPTION
**Pull request by nniranjhana**
Due to not checking for the right regex on console, the test was running into errors during automation. Fix it to pass without changing test behavior.

Add the check for pm_debug hooks (which was added by #10655) and enable configs and logging to see the output.

Add the check for lock/unlock PM states (which was added by #11083) and enable corresponding config.

Signed-off-by: Niranjhana N niranjhana.n@intel.com

### CODE REVIEW

File: samples/subsys/power/power_mgr/src/main.c [hunk 1]
```diff
@@ -40,7 +40,7 @@ void main(void)
 	printk("\n\n***OS Power Management Demo on %s****\n", CONFIG_ARCH);
 	printk(DEMO_DESCRIPTION);
 
-	for (int i = 1; i <= 4; i++) {
+	for (int i = 1; i <= 2; i++) {
```
[the remark this answers is not in the document]
Comment (nniranjhana) [reply samples/subsys/power/power_mgr/src/main.c]: Well we are using only two cases in the loop - odd and even so I changed it for clarity, it doesn't change the behavior, but if this was there for a reason, I'll change it back

[the remark this answers is not in the document]
Comment (nniranjhana) [reply samples/subsys/power/power_mgr/src/main.c]: Will change this back

File: samples/subsys/power/power_mgr/src/main.c [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <device.h>
 #include <gpio.h>
+#include "../subsys/power/policy/pm_policy.h"
```
[the remark this answers is not in the document]
Comment (nniranjhana) [reply samples/subsys/power/power_mgr/src/main.c]: Okay, done

**Review by ramakrishnapallala — APPROVED** [review]

**Comment by nniranjhana** [comment]
@nashif @andrewboie request to review, thanks

File: samples/subsys/power/power_mgr/CMakeLists.txt [hunk 3]
```diff
@@ -4,3 +4,4 @@ project(power_mgr)
 
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
+target_include_directories(app PRIVATE $ENV{ZEPHYR_BASE}/subsys/power)
```
Comment (nashif) [inline samples/subsys/power/power_mgr/CMakeLists.txt]: what exactly do you need from here? We should not include non-public APIs in samples.

File: samples/subsys/power/power_mgr/src/main.c [hunk 4]
```diff
@@ -11,6 +11,7 @@
 #include <string.h>
 #include <device.h>
 #include <gpio.h>
+#include "policy/pm_policy.h"
```
Comment (nashif) [inline samples/subsys/power/power_mgr/src/main.c]: hmm, is this a private API?

**Review by nashif — CHANGES_REQUESTED** [review]

File: samples/subsys/power/power_mgr/src/main.c [hunk 5]
```diff
@@ -56,5 +57,7 @@ void main(void)
 		}
 	}
 
+	sys_pm_dump_debug_info();
```
Comment (pizi-nordic) [inline samples/subsys/power/power_mgr/src/main.c]: Do we have any automated way to check if the tests works as expected?

File: samples/subsys/power/power_mgr/src/main.c (hunk 4 above)
Comment (ramakrishnapallala) [reply samples/subsys/power/power_mgr/src/main.c]: yes @nashif 

I will move this to include/power.h in my upcoming PR.

File: samples/subsys/power/power_mgr/src/main.c (hunk 5 above)
Comment (ramakrishnapallala) [reply samples/subsys/power/power_mgr/src/main.c]: as of now no, we might have to add new API or expend the existing one to return the details.

File: samples/subsys/power/power_mgr/src/main.c (hunk 4 above)
Comment (nashif) [reply samples/subsys/power/power_mgr/src/main.c]: can you submit that PR first?

File: samples/subsys/power/power_mgr/CMakeLists.txt (hunk 3 above)
Comment (nniranjhana) [reply samples/subsys/power/power_mgr/CMakeLists.txt]: I have removed these, now that Rama's #11656 is merged

**Review by ramakrishnapallala — APPROVED** [review]

**Comment by nashif** [comment]
obsolete sample

**Pull request closed without merging**